### PR TITLE
Add Policy Set Versions

### PR DESCRIPTION
**Pull request by omarismail**
## Description

This PR introduces support for [Policy Set Versions](https://www.terraform.io/docs/cloud/api/policy-sets.html#create-a-policy-set-version). The functionality allows for
* Creating Policy Set Versions
* Reading Policy Set Versions
* Uploading policies

## Output from tests

```
$ go-tfe % TFE_TOKEN=<token> TFE_ADDRESS=https://tfe-zone-4c292d5b.ngrok.io go test -v -run TestPolicySetVersion -timeout=30m
=== RUN   TestPolicySetVersionsCreate
=== RUN   TestPolicySetVersionsCreate/with_valid_identifier
=== RUN   TestPolicySetVersionsCreate/with_invalid_identifier
--- PASS: TestPolicySetVersionsCreate (1.49s)
    --- PASS: TestPolicySetVersionsCreate/with_valid_identifier (0.22s)
    --- PASS: TestPolicySetVersionsCreate/with_invalid_identifier (0.00s)
=== RUN   TestPolicySetVersionsRead
=== RUN   TestPolicySetVersionsRead/with_valid_id
=== RUN   TestPolicySetVersionsRead/with_invalid_id
--- PASS: TestPolicySetVersionsRead (1.86s)
    --- PASS: TestPolicySetVersionsRead/with_valid_id (0.17s)
    --- PASS: TestPolicySetVersionsRead/with_invalid_id (0.14s)
=== RUN   TestPolicySetVersionsUpload
=== RUN   TestPolicySetVersionsUpload/with_valid_upload_URL
=== RUN   TestPolicySetVersionsUpload/with_missing_upload_URL
--- PASS: TestPolicySetVersionsUpload (1.81s)
    --- PASS: TestPolicySetVersionsUpload/with_valid_upload_URL (0.42s)
    --- PASS: TestPolicySetVersionsUpload/with_missing_upload_URL (0.00s)
PASS
ok      github.com/hashicorp/go-tfe     5.773s
```

## External links

- [API documentation](https://www.terraform.io/docs/cloud/api/policy-sets.html#create-a-policy-set-version)
- [GitHub Issue](https://github.com/hashicorp/go-tfe/issues/151)

### CODE REVIEW

File: policy_set_version.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	Upload(ctx context.Context, psv PolicySetVersion, path string) error
+}
+
+// policySetVersions implements PolicySetVersions.
+type policySetVersions struct {
+	client *Client
+}
+
+// PolciySetVersionSource represents a source type of a policy set version.
+type PolciySetVersionSource string
```
Comment (chrisarcand) [inline policy_set_version.go:37]: For this and several other places in these changes, `s/ Polciy/Policy/` 😄  I'm impressed by the consistency, though!

File: policy_set_version.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+)
+
+// PolicySetVersionStatus represents a policy set version status.
+type PolicySetVersionStatus string
+
+//List all available policy set version statuses.
+const (
+	PolicySetVersionErrored PolicySetVersionStatus = "errored"
+	PolicySetVersionPending PolicySetVersionStatus = "pending"
+	PolicySetVersionReady   PolicySetVersionStatus = "ready"
```
Comment (chrisarcand) [inline policy_set_version.go:55]: `ingressing` is a potential status that seems to be missing here.

File: policy_set_version.go [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+	ReadyAt      time.Time `jsonapi:"attr,ready-at,rfc3339"`
+	ErroredAt    time.Time `jsonapi:"attr,errored-at,rfc3339"`
+}
+
+type PolicySetVersion struct {
+	ID               string                           `jsonapi:"primary,policy-set-versions"`
+	Source           PolciySetVersionSource           `jsonapi:"attr,source"`
+	Status           PolicySetVersionStatus           `jsonapi:"attr,status"`
+	StatusTimestamps PolciySetVersionStatusTimestamps `jsonapi:"attr,status-timestamps"`
+	Error            string                           `jsonapi:"attr,error"`
```
Comment (chrisarcand) [inline policy_set_version.go:72]: Missing `error-message` (might be missing from the docs, too).

File: tfe.go [hunk 4]
```diff
@@ -743,3 +747,22 @@ func checkResponseCode(r *http.Response) error {
 
 	return fmt.Errorf(strings.Join(errs, "\n"))
 }
+
+func readFile(path string) (*bytes.Buffer, error) {
```
Comment (chrisarcand) [inline tfe.go:751]: `readFile` doesn't feel right; Can we convey that we're actually packing up a file/path somehow? `packContents(path string)`, something? 🤷🏻‍♂️

**Review by chrisarcand — CHANGES_REQUESTED** [review]

File: policy_set_version.go (hunk 1 above)
Comment (omarismail) [reply policy_set_version.go:37]: Lots of vim copy/pasting 😅 . Will update

File: tfe.go (hunk 4 above)
Comment (omarismail) [reply tfe.go:751]: Yeah, this felt off but wasn't sure on the right naming. `packContents` could work. 🤔

**Review by chrisarcand — APPROVED** [review]
LGTM!

**Pull request merged**